src/MSONParameterParser.h

```diff
@@ -0,0 +1,221 @@
+
+                    break;
+                }
+
+                case MSONValueMembersSectionType:
+                {
+                    typeSection.node.baseType = mson::ImplicitValueBaseType;
+                    cur = MSONTypeSectionListParser::parse(node, siblings, pd, typeSection);
+
+                    out.node.values.clear();
+
+                    if (pd.exportSourceMap()) {
+                        out.sourceMap.values.collection.clear();
+                    }
+
+                    for (size_t i = 0; i < typeSection.node.content.elements().size(); i++) {
+                        mson::ValueMember valueMember = typeSection.node.content.elements().at(i).content.value;
+                        SourceMap<mson::ValueMember> valueMemberSM;
+
+                        if (pd.exportSourceMap()) {
+                            valueMemberSM = typeSection.sourceMap.elements().collection.at(i).value;
+                        }
+
+                        if (valueMember.valueDefinition.values.size() == 1) {
+                            out.node.values.push_back(valueMember.valueDefinition.values[0].literal);
+
+                            if (pd.exportSourceMap()) {
+                                SourceMap<Value> valueSM;
+                                valueSM.sourceMap = valueMemberSM.valueDefinition.sourceMap;
+
+                                out.sourceMap.values.collection.push_back(valueSM);
+                            }
+                        }
+                    }
+
+                    break;
+                }
+                    
+                default:
+                    break;
+            }
+
+            return cur;
+        }
+
+        static void finalize(const MarkdownNodeIterator& node,
+                             SectionParserData& pd,
+                             const ParseResultRef<MSONParameter>& out) {
+
+            SectionProcessor<Parameter>::checkDefaultAndRequiredClash<MSONParameter>(node, pd, out);
+
+            if ((!out.node.exampleValue.empty() || !out.node.defaultValue.empty()) &&
+                !out.node.values.empty()) {
+
+                SectionProcessor<Parameter>::checkExampleAndDefaultValue<MSONParameter>(node, pd, out);
+            }
+        }
+
+        static SectionType nestedSectionType(const MarkdownNodeIterator& node) {
+
+            SectionType nestedType = UndefinedSectionType;
+
+            // Recognize `Default` and `Members` sections
+            nestedType = SectionProcessor<mson::TypeSection>::sectionType(node);
+
+            return nestedType;
+        }
+
+        static void parseAttributes(const MarkdownNodeIterator& node,
+                                    SectionParserData& pd,
+                                    const std::vector<mdp::ByteBuffer>& attributes,
+                                    const ParseResultRef<MSONParameter>& out) {
+
+            out.node.use = UndefinedParameterUse;
+
+            if (attributes.size() <= 2) {
```

Comment:
@pksunkara why this magic number here (hint: please use constants with an explanatory comment)